pkg/nimble/patches/0001-treewide-fix-compilation-with-clang.patch

```diff
@@ -0,0 +1,289 @@
+From 4bac07eb6c80d2f69a1ebb094cb7e1e9c6f85b3a Mon Sep 17 00:00:00 2001
+From: Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+Date: Sat, 20 May 2023 21:27:57 +0200
+Subject: [PATCH] treewide: fix compilation with clang
+
+---
+ nimble/controller/src/ble_ll_adv.c           | 1 +
+ nimble/controller/src/ble_ll_conn.c          | 1 +
+ nimble/controller/src/ble_ll_scan.c          | 3 +++
+ nimble/controller/src/ble_ll_scan_aux.c      | 3 +++
+ nimble/drivers/nrf52/src/ble_phy.c           | 1 +
+ nimble/host/services/gap/src/ble_svc_gap.c   | 2 ++
+ nimble/host/services/gatt/src/ble_svc_gatt.c | 2 ++
+ nimble/host/services/ipss/src/ble_svc_ipss.c | 2 ++
+ nimble/host/src/ble_att_clt.c                | 1 +
+ nimble/host/src/ble_gap.c                    | 2 +-
+ nimble/host/src/ble_hs.c                     | 2 ++
+ nimble/host/src/ble_hs_conn.c                | 2 ++
+ nimble/host/src/ble_l2cap.c                  | 1 +
+ nimble/transport/src/transport.c             | 5 +++++
+ porting/nimble/src/os_msys_init.c            | 4 +++-
+ 15 files changed, 30 insertions(+), 2 deletions(-)
+
+diff --git a/nimble/controller/src/ble_ll_adv.c b/nimble/controller/src/ble_ll_adv.c
+index 3327ba5..1fdb439 100644
+--- a/nimble/controller/src/ble_ll_adv.c
++++ b/nimble/controller/src/ble_ll_adv.c
+@@ -1512,6 +1512,7 @@ ble_ll_adv_aux_schedule_next(struct ble_ll_adv_sm *advsm)
+     BLE_LL_ASSERT(AUX_DATA_LEN(advsm) >= next_aux_data_offset);
+ 
+     rem_aux_data_len = AUX_DATA_LEN(advsm) - next_aux_data_offset;
++    (void)rem_aux_data_len;
```

Comment:
can't we do

    CFLAGS += -Wno-unused-parameter

Reply:
It's not an unused parameter, but an unused variable.

Reply:
Yea but isn't there a [switch for that](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wno-unused-variable) too?